packages/workflow/src/internals.ts

```diff
@@ -785,6 +790,41 @@ export class Activator implements ActivationHandler {
     this.knownPresentPatches.add(activation.patchId);
   }
 
+  public patchInternal(patchId: string, deprecated: boolean): boolean {
+    if (this.workflow === undefined) {
+      throw new IllegalStateError('Patches cannot be used before Workflow starts');
+    }
+    const usePatch = !this.info.unsafe.isReplaying || this.knownPresentPatches.has(patchId);
+    // Avoid sending commands for patches core already knows about.
+    // This optimization enables development of automatic patching tools.
+    if (usePatch && !this.sentPatches.has(patchId)) {
+      this.pushCommand({
+        setPatchMarker: { patchId, deprecated },
+      });
+      this.sentPatches.add(patchId);
+    }
+    return usePatch;
+  }
+
+  // Called early while handling an activation to register known flags
+  public addKnownFlags(flags: number[]): void {
+    for (const flag of flags) {
+      assertValidFlag(flag);
+      this.knownFlags.add(flag);
+    }
+  }
+
+  public hasFlag(flag: SdkFlag): boolean {
+    if (this.knownFlags.has(flag.id)) {
+      return true;
+    }
+    if (!this.info.unsafe.isReplaying && flag.default) {
```

Comment:
Looks like `default` is footgun since we wouldn't add it to the set of known flags by default.
I recommend removing that layer of complexity and assume all flags are off by default.

Reply:
See my other comment.